src/strategies_and_params/reportingstrategy.jl

```diff
@@ -225,14 +241,14 @@ function refine_r_strat(s::ReportToFile)
     end
     return s
 end
-function report!(s::ReportToFile, _, args...)
+function report!(s::ReportToFile, step, args...)
     if s.save_if
-        report!(args...)
+        step % s.reporting_interval == 0 && report!(args...)
     end
     return nothing
 end
```

Comment:
Again, I the way I understand it is that the modulo condition is already checked before this code is called. Restoring the previous version of the code seems fine.

Reply:
You are right, this check is redundant